### PR TITLE
[CBRD-24887] Register mmon utility

### PR DESCRIPTION
**Pull request by Rudeus**
http://jira.cubrid.org/browse/CBRD-24887

Register mmon utility

### CODE REVIEW

File: src/executables/utility.h [hunk 1]
```diff
@@ -1691,6 +1693,16 @@ typedef struct _ha_config
 #define FLASHBACK_OLDEST_S          14102
 #define FLASHBACK_OLDEST_L          "oldest"
 
+/* mmon option list */
+#define MMON_MODULE_S         'm'
+#define MMON_MODULE_L         "20123"
```
Comment (wrlawodms) [inline src/executables/utility.h]:
```suggestion
#define MMON_MODULE_L         20123
```
You may want to do this?

File: src/executables/util_admin.c [hunk 2]
```diff
@@ -893,6 +893,23 @@ static GETOPT_LONG ua_Flashback_Option[] = {
...
+  {MMON_TRANSACTION_S, {ARG_BOOLEAN}, {0}},
+  {MMON_TRAN_COUNT_S, {ARG_INTEGER}, {0}},
+  {MMON_SHOW_ALL_S, {ARG_BOOLEAN}, {0}},
+  {0, {0}, {0}}
+};
+
+static GETOPT_LONG ua_Mmon_Option[] = {
+  {MMON_MODULE_L, 1, 0, MMON_MODULE_S},
+  {MMON_TRANSACTION_L, 0, 0, MMON_TRANSACTION_S},
+  {MMON_TRAN_COUNT_L, 0, 0, MMON_TRAN_COUNT_S},
```
Comment (wrlawodms) [inline src/executables/util_admin.c]:
```suggestion
  {MMON_TRAN_COUNT_L, 1, 0, MMON_TRAN_COUNT_S},
```

File: src/executables/utility.h (hunk 1 above)
Comment (Rudeus) [reply src/executables/utility.h]: Exchange to long-term

File: src/executables/util_admin.c (hunk 2 above)
Comment (Rudeus) [reply src/executables/util_admin.c]: Done.

File: src/executables/utility.h [hunk 3]
```diff
@@ -1691,6 +1693,16 @@ typedef struct _ha_config
 #define FLASHBACK_OLDEST_S          14102
 #define FLASHBACK_OLDEST_L          "oldest"
 
+/* mmon option list */
+#define MMON_MODULE_S         'm'
+#define MMON_MODULE_L         "module"
+#define MMON_TRANSACTION_S    't'
+#define MMON_TRANSACTION_L    "trnasction"
```
Comment (joohok) [inline src/executables/utility.h]:
```suggestion
#define MMON_TRANSACTION_L    "transaction"
```

File: src/executables/util_cs.c [hunk 4]
```diff
@@ -4540,3 +4540,21 @@ flashback (UTIL_FUNCTION_ARG * arg)
...
+  bool transaction, show_all;
+  int tran_count;
+  module = utility_get_option_string_value (arg_map, MMON_MODULE_S, 0);
+  transaction = utility_get_option_bool_value (arg_map, MMON_TRANSACTION_S);
+  tran_count = utility_get_option_int_value (arg_map, MMON_TRAN_COUNT_S);
+  show_all = utility_get_option_bool_value (arg_map, MMON_SHOW_ALL_S);
+
+  // XXX: for test, it will removed at main implementation
+  fprintf (stdout, "mmon utility: -m %s, -t %s, -c %d, -a %s\n", (module ? module : "NULL"),
+	   (transaction ? "true" : "false"), tran_count, (show_all ? "true" : "false"));
```
Comment (wrlawodms) [inline src/executables/util_cs.c:4558]:
```suggestion
	   transaction, tran_count, show_all);
```
but it's just test code. Who cares.

**Review by joohok — APPROVED** [review]

File: src/executables/utility.h (hunk 3 above)
Comment (wrlawodms) [inline src/executables/utility.h]:
```suggestion
#define MMON_TRANSACTION_L    "transaction"
```

**Review by wrlawodms — APPROVED** [review]

File: src/executables/utility.h [hunk 5]
```diff
@@ -997,6 +998,7 @@ typedef struct _ha_config
 #define UTIL_OPTION_CHECKSUMDB			"checksumdb"
 #define UTIL_OPTION_TDE			        "tde"
 #define UTIL_OPTION_FLASHBACK                   "flashback"
+#define UTIL_OPTION_MMON                        "mmon"
```
Comment (hornetmj) [inline src/executables/utility.h]: It would be better memmon.

Comment (Rudeus) [reply src/executables/utility.h]: Done.

File: src/executables/utility.h [hunk 6]
```diff
@@ -1691,6 +1693,16 @@ typedef struct _ha_config
 #define FLASHBACK_OLDEST_S          14102
 #define FLASHBACK_OLDEST_L          "oldest"
 
+/* mmon option list */
```
Comment (joohok) [inline src/executables/utility.h]:
```suggestion
/* memmon option list */
```

File: src/executables/util_cs.c [hunk 7]
```diff
@@ -4540,3 +4540,21 @@ flashback (UTIL_FUNCTION_ARG * arg)
...
 
 }
+
+int
+memmon (UTIL_FUNCTION_ARG * arg)
+{
+  UTIL_ARG_MAP *arg_map = arg->arg_map;
+  char *module;
+  bool transaction, show_all;
+  int tran_count;
```
Comment (hornetmj) [inline src/executables/util_cs.c:4550]:
```suggestion
  int tran_count;
  
```

**Review by hornetmj — APPROVED** [review]

File: src/executables/util_cs.c (hunk 7 above)
Comment (hornetmj) [reply src/executables/util_cs.c:4550]: New line after end of the declaration section of local variables.

**Pull request merged**